src/util/number_utils.ts

```diff
@@ -0,0 +1,20 @@
+import { BigNumber } from '0x.js';
+
+export const padRightSplitted = (numBg: BigNumber, decimals: number = 4): { num: string; diff: string } => {
+    const numBgToFixed = numBg.toFixed(decimals);
+    const numBgToString = numBg.toString();
+
+    const decimalPlaces = (numBgToString.split('.')[1] || []).length;
+
+    let diff = '';
+    let num = numBgToFixed;
+    if (!numBg.isZero() && decimalPlaces < decimals) {
+        diff = numBgToFixed.split(numBgToString).join('');
```

Comment:
This is equivalent to `.replace(numBgToString, '')`, isn't it?

Reply:
Yes @fvictorio , Do you think the replace is a better option?

Reply:
Yes

Reply:
Solved